src/pointer.cr

```diff
@@ -491,6 +491,31 @@ struct Pointer(T)
     Slice.new(self, size)
   end
 
+  # Returns a `Slice` from a given null-terminated pointer array, with an optional,
+  # though recommended, limit.
+  #
+  # ```
+  # ptr = LibC.getgrnam("wheel").value.gr_mem
+  # slice = pointer.to_null_terminated_slice(LibC::NGROUP_MAX)
+  # ```
+  def to_slice_null_terminated(limit : Int)
```

Comment:
I'm not sure we need these methods. Maybe improve `String.new` to take a maximum size argument, as an alternative to the fixed size argument we have now —I got this use case a few times, for example recently while parsing the section name of PE files (windows executable).

Reply:
It doesn't necessarily produce a string. If you have Type** which is a null terminated array. This will only deal with the array giving you a Slice(Type*).

Reply:
The use case for this is a null terminated array of things. I thong think this has anything to do with `String`. As far as I can tell there is nothing that fills this use case in the stdlib. What would you have me replace it with? The only `new` that might work would be something for `Array` or `Slice` like this: 
```
def initialize(ptr : Pointer(T), limit : Int)
  # ...
end

def initialize(ptr : Pointer(T))
  # ...
end
```
Thoughts?